### PR TITLE
PostgreSQL Flexible Server GA Swagger API changes 

### PR DESCRIPTION
**Pull request by xunsun-commits**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  2021 May.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  2021 June.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  },
+  "responses": {
+    "201": {
+      "body": {
+        "sku": {
+            "name": "Standard_D4s_v3",
+            "tier": "GeneralPurpose"
+        },
+        "properties": {
+            "fullyQualifiedDomainName": "c7d7483a8ceb.test-private-dns-zone.postgres.database.azure.com",
```
Comment (DaeunYim) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json:47]: shouldn't it be `pgtestsvc4.postgres.database.azure.com` ??

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      }
+    },
+    "200": {
+      "body": {
+        "sku": {
+            "name": "Standard_D4s_v3",
+            "tier": "GeneralPurpose"
+        },
+        "properties": {
+            "fullyQualifiedDomainName": "c7d7483a8ceb.test-private-dns-zone.postgres.database.azure.com",
```
Comment (DaeunYim) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json:94]: Same incorrect domain name ` pgtestsvc4.postgres.database.azure.com`

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "sku": {
+            "name": "Standard_D4s_v3",
+            "tier": "GeneralPurpose"
+        },
+        "properties": {
+            "fullyQualifiedDomainName": "c7d7483a8ceb.test-private-dns-zone.meru.com",
```
Comment (DaeunYim) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json:16]: Same incorrect domain name ` pgtestsvc4.postgres.database.azure.com` and in other examples

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            },
+            "backup": {
+                "backupRetentionDays": 7,
+                "geoRedundantBackup": "Disabled",
+                "earliestRestoreDate": "2021-05-26T01:16:58.3723361+00:00"
+            },
+            "network": {
+                "publicNetworkAccess": "Disabled",
+                "delegatedSubnetResourceId": "/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/testrg/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/test-vnet-subnet",
+                "privateDnsZoneArmResourceId": "/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/testrg/providers/Microsoft.Network/privateDnsZones/test-private-dns-zone.meru.com"
```
Comment (DaeunYim) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json:33]: Is this meru.com dns zone suffix supported? I think we only support ".postgres.database.azure.com"

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json (hunk 3 above)
Comment (taogems) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json:16]: This probably needs to end with postgres.database.azure.com just to be consistent.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json (hunk 4 above)
Comment (taogems) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json:33]: Same here.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerList.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      "body": {
+        "nextLink": "https://management.azure.com/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/providers/Microsoft.DBforPostgreSQL/flexibleServers?api-version=2021-06-01-preview&$top=3&$skiptoken=skiptoken",
+        "value": [
+          {
+              "sku": {
+                  "name": "Standard_D4s_v3",
+                  "tier": "GeneralPurpose"
+              },
+              "properties": {
+                  "fullyQualifiedDomainName": "c7d7483a8ceb.test-private-dns-zone.meru.com",
```
Comment (taogems) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerList.json:17]: Please make sure we don't have meru.com in this PR. meru.com is just for our onebox.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json [hunk 6]
```diff
@@ -0,0 +1,1885 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
```
Comment (yannickzj) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json:349]: Can we add a body parameter to include restart parameters? HA servers need this to trigger forced/planned failover.

example change: https://github.com/Azure/azure-rest-api-specs/pull/14389/commits/905c382f24bf0906f63685176fdc8f00b8bbcf6d#diff-13064bc232d88ad907503b5f99792fe98379706b201c051b4d72e417b1f035ff

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/postgresql.json [hunk 7]
```diff
@@ -0,0 +1,1885 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
```
Comment (yannickzj) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/postgresql.json:349]: same here, can we add a body parameter to include restart parameters?

**Review by yannickzj — COMMENTED** [review]
would be great if you can add an example for server restart with failover as well :)

example change: https://github.com/Azure/azure-rest-api-specs/commit/905c382f24bf0906f63685176fdc8f00b8bbcf6d#diff-604724e22ec2150f20ddccaa86c38597e54cc436383bf3a2fb15025cfec993bc

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json (hunk 6 above)
Comment (xunsun-commits) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json:349]: Thanks for reminding, added.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/postgresql.json (hunk 7 above)
Comment (xunsun-commits) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/postgresql.json:349]: added

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json (hunk 3 above)
Comment (xunsun-commits) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json]: Thanks, changed

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json (hunk 1 above)
Comment (xunsun-commits) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json]: Thanks. This is the scenario of VNet injection. So there won't be a public IP for server. Instead, the private DNS zone will be used to access server. Tao should know this better. The differences between public IP server and VNet injection server are shown in ServerGet.json and ServerGetWithVnet.json

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json (hunk 4 above)
Comment (xunsun-commits) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerGetWithVnet.json]: changed

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json (hunk 1 above)
Comment (DaeunYim) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-06-01-preview/examples/ServerCreate.json]: Thanks a lot Tiger for checking this!

**Comment by leni-msft** [comment]
@allenjzhang could you help check the .net sdk generation failure? thanks

**Comment by xunsun-commits** [comment]
Opened a same PR for better code review experience: https://github.com/Azure/azure-rest-api-specs/pull/14653

**Pull request closed without merging**